src/shared/Core.UI/ViewModels/CredentialsViewModel.cs

```diff
@@ -0,0 +1,66 @@
+using System.ComponentModel;
+
+namespace GitCredentialManager.UI.ViewModels
+{
+    public class CredentialsViewModel : WindowViewModel
+    {
+        private string _userName;
+        private string _password;
+        private string _description;
+        private bool _showProductHeader;
+        private RelayCommand _signInCommand;
+
+        public CredentialsViewModel()
+        {
+            SignInCommand = new RelayCommand(Accept, CanSignIn);
+            PropertyChanged += OnPropertyChanged;
+        }
+
+        private void OnPropertyChanged(object sender, PropertyChangedEventArgs e)
+        {
+            switch (e.PropertyName)
+            {
+                case nameof(UserName):
+                case nameof(Password):
+                    SignInCommand.RaiseCanExecuteChanged();
+                    break;
+            }
+        }
+
+        private bool CanSignIn()
+        {
+            // Allow empty username or empty password, but not both
+            return !string.IsNullOrWhiteSpace(UserName) || !string.IsNullOrEmpty(Password);
```

Comment:
This is technically a difference from the Windows API that this replaces; the Windows dialog allows you to submit an empty username _and_ password.

Do you think we should remove all validation here @ldennington ?

Reply:
If I'm understanding correctly (i.e. this is no longer applicable) than I'm fine with removing!

Reply:
Might be safer to let users continue to submit empty user/pass.